app/scripts/session/sessionServices.js

```diff
@@ -30,48 +28,47 @@ window.vcSession = window.vcSession ? window.vcSession : {};
 
     return {
       saveSession: function(virtualCohort) {
-        var data = {
-          virtualCohort: virtualCohort
-        };
         $.ajax({
           type: 'POST',
           url: vcSession.URL,
           contentType: 'application/json;charset=UTF-8',
-          data: JSON.stringify(data)
+          data: JSON.stringify(virtualCohort)
         }).done(function(response) {
-          if (virtualCohort.userID === 'DEFAULT')
-            localStorageAdd_(response.id,
-              virtualCohort);
+          if (virtualCohort.userID === 'DEFAULT') {
+            virtualCohort.virtualCohortID = response.id;
+            localStorageAdd_(virtualCohort);
+          }
         }).fail(function() {
-          localStorageAdd_(vcSession.utils.generateUUID(),
-            virtualCohort);
+          virtualCohort.virtualCohortID = vcSession.utils.generateUUID();
+          localStorageAdd_(virtualCohort);
         });
       },
       removeSession: function(_virtualCohort) {
         $.ajax({
           type: 'DELETE',
-          url: vcSession.URL + _virtualCohort.virtualCohortID,
+          url: vcSession.URL + '/' + _virtualCohort.virtualCohortID,
           contentType: 'application/json;charset=UTF-8'
         }).done(function() {
-          if (_virtualCohort.userID === 'DEFAULT')
+          if (_virtualCohort.userID === 'DEFAULT') {
             localStorageDelete_(_virtualCohort);
+          }
         }).fail(function() {
           localStorageDelete_(_virtualCohort);
         });
       },
       editSession: function(_virtualCohort) {
-        var data = {
-          virtualCohort: _virtualCohort
-        };
         $.ajax({
           type: 'PUT',
-          url: vcSession.URL + _virtualCohort.virtualCohortID,
+          url: vcSession.URL + '/' + _virtualCohort.virtualCohortID,
           contentType: 'application/json;charset=UTF-8',
-          data: JSON.stringify(data)
-        }).done(function(response) {
-          if (_virtualCohort.userID === 'DEFAULT')
-            localStorageEdit_(data.virtualCohort);
+          data: JSON.stringify(_virtualCohort)
+        }).done(function() {
+          if (_virtualCohort.userID === 'DEFAULT') {
+            localStorageEdit_(_virtualCohort);
+          }
         }).fail(function(jqXHR) {
+          // TODO: should we delete the virtual cohort if no record found
+          // in the database? Should we add it into database?
```

Comment:
Why dele from local storage and saveSession at the same time? And I think if the session service is not running, or for whatever reason we can't retrieve the VC from session service, we should leave everything as it in the drop down, but when user click on visualize/query just shows an error message